iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergTableOptimizer.java

```diff
@@ -115,27 +134,19 @@ private List<org.apache.hadoop.hive.common.TableName> getTables() {
     }
   }
 
-  private List<String> getPartitions(org.apache.iceberg.Table icebergTable) {
-    try {
-      return IcebergTableUtil.getPartitionNames(icebergTable, Maps.newHashMap(), true);
-    } catch (SemanticException e) {
-      throw new RuntimeMetaException(e, "Error getting partition names for Iceberg table %s", icebergTable.name());
-    }
-  }
-
-  private Table resolveMetastoreTable(String qualifiedTableName) {
-    String[] dbTableName = TxnUtils.getDbTableName(qualifiedTableName);
+  private org.apache.hadoop.hive.ql.metadata.Table getHiveTable(String dbName, String tableName) {
     try {
-      return metadataCache.computeIfAbsent(qualifiedTableName,
-          () -> CompactorUtil.resolveTable(conf, dbTableName[0], dbTableName[1]));
+      Table metastoreTable = metadataCache.computeIfAbsent(TableName.getDbTable(dbName, tableName), () ->
+          CompactorUtil.resolveTable(conf, dbName, tableName));
+      return new org.apache.hadoop.hive.ql.metadata.Table(metastoreTable);
     } catch (Exception e) {
-      throw new RuntimeMetaException(e, "Error resolving table %s", qualifiedTableName);
+      throw new RuntimeMetaException(e, "Error getting Hive table");
```

Comment:
could we keep the table name in the exception

Reply:
Done